pkg/virt-controller/watch/snapshot/restore.go

```diff
@@ -343,8 +349,13 @@ func (t *vmRestoreTarget) Reconcile() (bool, error) {
 	var deletedDataVolumes []string
 	updatedStatus := false
 
-	copy(newTemplates, snapshotVM.Spec.DataVolumeTemplates)
-	copy(newVolumes, snapshotVM.Spec.Template.Spec.Volumes)
+	for i, t := range snapshotVM.Spec.DataVolumeTemplates {
```

Comment:
this was actually the problematic block.  `snapshotVM.Spec.Template.Spec.Volumes` was getting overwritten